### ISSUE TITLE
Linker error with CONFIG_NET_TCP=y

### ISSUE THREAD

**Issue by tfir1981** [platformio/platform-ststm32#573]
**Describe the bug**
I am using the NUCLEO_F412ZG Board and i want to setup an MQTT-Application with an w5500 mac-phy. Adding step by step my Kconfig-Parameters, i get a linker error after adding CONFIG_NET_TCP.

```properties
CONFIG_MAIN_STACK_SIZE=4096

CONFIG_CPLUSPLUS=y
CONFIG_LIB_CPLUSPLUS=y
CONFIG_STD_CPP11=y
CONFIG_NEWLIB_LIBC=y
CONFIG_NEWLIB_LIBC_NANO=n
CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y

CONFIG_SPI=y

CONFIG_ETH_W5500=y

CONFIG_ETHERNET_LOG_LEVEL_DBG=y

CONFIG_NETWORKING=y
CONFIG_NET_SHELL=y
CONFIG_NET_L2_ETHERNET=y

CONFIG_ENTROPY_GENERATOR=y
CONFIG_TEST_RANDOM_GENERATOR=y

# Networking Config:
CONFIG_NET_SOCKETS=y
CONFIG_NET_SOCKETS_POSIX_NAMES=n
CONFIG_NET_IPV4=y
CONFIG_NET_IPV6=n
CONFIG_NET_TCP=y

# Enable the MQTT Lib
CONFIG_MQTT_LIB=y

# Network address config
CONFIG_NET_CONFIG_SETTINGS=y
# CONFIG_NET_CONFIG_NEED_IPV4=y
CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.0.2.2"
```

`CONFIG_MQTT_LIB` implicitly sets `CONFIG_NET_SOCKETS=y`. So far so good. Without `CONFIG_NET_TCP` build is fine. But after turning this switch to `y` i get the linker error below.
Finally i was studying the documentation of [BSD Socket](https://docs.zephyrproject.org/2.6.0/reference/networking/sockets.html) and found some hints concerning libc. 
I build up my Kconfg with the help of the zephyr-samples. So i think `CONFIG_NET_TCP` is necessary for mqtt comminication. 

**Expected behavior**
Build without errors.
Building on Zephyr's 'west' is fine.

**Impact**
showstopper: Cannot build Mqtt-Samples

**Logs and console output**
```bash
Linking .pio/build/my_eval_board/zephyr/firmware-pre.elf
/Project/.piocore/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/8.2.1/../../../../arm-none-eabi/bin/ld.bfd: .piocore/packages/toolchain-gccarmnoneeabi/arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(lib_a-fini.o): in function `__libc_fini_array':
fini.c:(.text.__libc_fini_array+0x26): undefined reference to `_fini'
collect2: error: ld returned 1 exit status
*** [.pio/build/my_eval_board/zephyr/firmware-pre.elf] Error 1
```

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 20.04 LTS
 - Toolchain Zephyr 2.6.0 with PlatformIO

**Comment by valeros** [comment]
Hi @tfir1981 ! Could you please update the framework-zephyr package to the latest and try again? Just run `pio platform update ststm32` in your IDE terminal.

**Comment by tfir1981** [comment]
Issue is solved in Zephyr 2.7.1
